File: aquality-appium-java-master/src/test/java/test/TestSecretCloset.java
package test;

import base.BaseTest;
import models.Product;
import models.Seller;
import org.testng.annotations.Test;
import steps.MainMenuScreenSteps;
import steps.ProductScreenSteps;
import steps.SelectCityScreenSteps;
import steps.SellerScreenSteps;

import static enums.CityItem.ABUDHABI;

public class TestSecretCloset extends BaseTest {

    @Test()
    public void testSecretCloset() {
        MainMenuScreenSteps.clickBtnSelectCity();
        SelectCityScreenSteps.clickBtnOkOnAlertLabel();
        String firstAvailableCity = SelectCityScreenSteps.getNameFirstAvailableCity();
        SelectCityScreenSteps.selectCityByName(ABUDHABI);
        MainMenuScreenSteps.checkSelectCity(firstAvailableCity);
        Product firstProductWithDiscount = MainMenuScreenSteps.getFirstProductWithDiscount();
        MainMenuScreenSteps.clickFirstProductWithDiscount();
        ProductScreenSteps.checkProductPricesAndDiscount(firstProductWithDiscount);
        Seller sellerFromProductScreen = ProductScreenSteps.getSellerFromProductScreen();
        ProductScreenSteps.clickNameSeller();
        SellerScreenSteps.checkSeller(sellerFromProductScreen);
    }
}

File: KotlinSecretCloset/src/test/java/test/TestSecretCloset.kt
package test

import base.BaseTest
import enums.CityItem
import org.testng.annotations.Test
import steps.*

class TestSecretCloset : BaseTest() {

    @Test
    fun testSecretCloset() {
        clickBtnSelectCity()
        clickBtnOkOnAlertLabel()
        selectCityByName(CityItem.ABUDHABI)
        checkSelectCity(CityItem.ABUDHABI.city!!)
        getFirstProductWithDiscount().let {
            clickFirstProductWithDiscount()
            checkProductPricesAndDiscount(it)
        }
        getSellerFromProductScreen().let {
            clickNameSeller()
            checkSeller(it)
        }
    }
}